frontend/src/main/scala/bloop/dap/BloopDebugToolsResolver.scala

```diff
@@ -0,0 +1,70 @@
+package bloop.dap
+
+import java.net.URLClassLoader
+
+import scala.collection.mutable
+import scala.util.Success
+import scala.util.Try
+
+import ch.epfl.scala.debugadapter.BuildInfo
+import ch.epfl.scala.debugadapter.DebugToolsResolver
+import ch.epfl.scala.debugadapter.ScalaVersion
+
+import bloop.DependencyResolution
+import bloop.DependencyResolution.Artifact
+import bloop.io.AbsolutePath
+import bloop.logging.Logger
+
+/**
+ * Resolve the tools for the debugger in the form of classloaders.
+ * The debug tools are comprised of:
+ * - the expression compiler
+ * - the step filter
+ */
+class BloopDebugToolsResolver(logger: Logger) extends DebugToolsResolver {
+  import BloopDebugToolsResolver._
+  override def resolveExpressionCompiler(scalaVersion: ScalaVersion): Try[ClassLoader] = {
+    getOrTryUpdate(expressionCompilerCache, scalaVersion) {
+      val module = s"${BuildInfo.expressionCompilerName}_$scalaVersion"
+      val artifact = Artifact(BuildInfo.organization, module, BuildInfo.version)
+      DependencyResolution
+        .resolveWithErrors(List(artifact), logger)
+        .map(jars => toClassLoader(jars, false))
+        .toTry
+    }
+  }
+
+  override def resolveStepFilter(scalaVersion: ScalaVersion): Try[ClassLoader] = {
+    getOrTryUpdate(stepFilterCache, scalaVersion) {
+      val stepFilterModule = s"${BuildInfo.scala3StepFilterName}_${scalaVersion.binaryVersion}"
+      val stepFilter = Artifact(BuildInfo.organization, stepFilterModule, BuildInfo.version)
+      val tastyCore = Artifact("org.scala-lang", "tasty-core_3", scalaVersion.value)
```

Comment:
Why do we need scala 3 artifacts on the classpath. Do we need it even with Scala 2?

Reply:
We only need to load the step filter for Scala 3.

The Scala 2 step filter is a direct dependency of the `scala-debug-adapter` so it does not need to be class loaded in its own class loader.